shared/hooks/useDeleteResource.js

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Button, Checkbox, MessageBox, MessageStrip } from 'fundamental-react';
+import LuigiClient from '@luigi-project/client';
+
+import {
+  useDelete,
+  useNotification,
+  navigateToList,
+  prettifyNameSingular,
+} from '..';
+import { useFeatureToggle } from '.';
+
+export function useDeleteResource({ i18n, resourceType }) {
+  const { t } = useTranslation(['translation'], { i18n });
+
+  const [showDeleteDialog, setShowDeleteDialog] = useState(false);
+  const deleteResourceMutation = useDelete();
+  const [dontConfirmDelete, setDontConfirmDelete] = useFeatureToggle(
+    'dontConfirmDelete',
+  );
+
+  const notification = useNotification();
+
+  const prettifiedResourceName = prettifyNameSingular(undefined, resourceType);
+
+  const performDelete = async resourceUrl => {
+    const withoutQueryString = path => path.split('?')[0];
+    const url = withoutQueryString(resourceUrl);
+
+    LuigiClient.sendCustomMessage({
+      id: 'busola.dontConfirmDelete',
+      value: dontConfirmDelete,
+    });
+    try {
+      await deleteResourceMutation(url);
+      notification.notifySuccess({
+        content: t('components.resources-list.messages.delete.success', {
+          resourceType: prettifiedResourceName,
+        }),
+      });
+      navigateToList(resourceType);
```

Comment:
it doesn't work well if we delete a resource from a list that's on a details page. for example - go to the details of a function, and delete a pod/api rule that's connected to it. it should stay in the same view, yet we're redirected to the list of functions.